### PR TITLE
BUG: optimize: Fix for 10303

### PR DESCRIPTION
**Pull request by mdhaber**
Starting with a dummy commit to test issue

### CODE REVIEW

**Comment by jwd0023** [comment]
Thanks for the change! Good work.

**Comment by ev-br** [comment]
Merged it quickly so that CI is green again. Thanks Matt.

(On a side note, I'm sure there are reasons for this upstream change, but nonetheless. The change from `mask ==False` to `logical_or(atleast_1d(...))` is not very intuitive)

**Comment by mdhaber** [comment]
@ev-br I was actually trying to _improve_ the style.
PEP8: 
"Don't compare boolean values to True or False using =="
```
Yes:   if greeting:
No:    if greeting == True:
Worse: if greeting is True:
```
I know they weren't thinking of arrays when they wrote it, but I still thought the sentiment was reasonable.
Disagree?

**Comment by ev-br** [comment]
That stanza is I think not applicable here. My grumble was that `... == 42`, is a clear and idiomatic way to spell "broadcast if necessary". The other one requires reading and thinking. Maybe it simply takes a bit to get used to :-).

**Comment by mdhaber** [comment]
Ok. I'll keep it in mind for the future.

**Comment by ev-br** [comment]
Nah, that was purely personal preference, feel free to disregard. And apologies if I sounded harsh, no offence is meant.

**Comment by mdhaber** [comment]
No problem. Good to know the preferences of people with experience!

**Comment by tylerjereddy** [comment]
I had to dissect some git history to find this as the likely fix for one of the linprog CI failures observed in  #10496 -- adding backport label..

File: scipy/optimize/_linprog_simplex.py [hunk 1]
```diff
@@ -90,7 +90,8 @@ def _pivot_col(T, tol=1e-9, bland=False):
     if ma.count() == 0:
         return False, np.nan
     if bland:
-        return True, np.nonzero(ma.mask == False)[0][0]
+        # ma.mask is sometimes 0d
+        return True, np.nonzero(np.logical_not(np.atleast_1d(ma.mask)))[0][0]
     return True, np.ma.nonzero(ma == ma.min())[0][0]
```
Comment (eric-wieser) [inline scipy/optimize/_linprog_simplex.py]: I think what you're actually looking for here is:
```
if ma.mask is np.ma.nomask:
    return True, 0
if bland:
    return True, np.nonzero(ma.mask == False)[0][0]
return True, np.ma.nonzero(ma == ma.min())[0][0]
```

As written in your, line 95 has the same issue

**Pull request merged**